DIPs/DIP1NNN-DK.md

```diff
@@ -0,0 +1,775 @@
+
+```D
+int* getPtr() @system {return cast(int*) 0x8035FDF0;}
+int  getVal() @system {return -1;}
+
+extern int* x0;                   // @safe by default
+int* x1 = x0;                     // @safe, (() => x0) is @safe
+int* x2 = cast(int*) 0x8035FDF0;  // @system, (() => cast(int*) 0x8035FDF0) is @system
+int* x3 = getPtr();               // @system, (() => getPtr()) is @system
+int  x4 = getVal();               // @safe, int is not an unsafe type
+@system int x5 = 1;               // @system as requested
+@trusted int* x6 = getPtr();      // @safe, the getPtr call gets trusted
+@safe int* x7 = getPtr();         // error: cannot initialize @safe variable with @system initializer
+
+struct S {
+    // same rules for fields:
+    int* x9 = x3; // @system
+    int  x8 = x5; // @safe
+}
+```
+
+An exception to the last rules is made on unsafe types when the compiler knows the resulting value is safe.
+```D
+int* getNull() pure @system {return null;}
+int* n = getNull(); // despite unsafe type with @system initialization expression, inferred as @safe
+```
+
+Annotations with a scope (`@system {}`) or colon (`@system:`) affect variables just like they do functions.
+```D
+@system {
+    int y0; // @system
+}
+
+@system:
+int y1; // @system
+```
+
+**(4) `__traits(getFunctionAttributes)` may be called on variables and fields**
+
+Currently it is possible to give function attributes to declarations that aren't functions.
+It is not possible however to inspect any of them.
+
+```D
+@system @nogc pure nothrow int x;
+pragma(msg, __traits(getFunctionAttributes, x)); // error: first argument is not a function
+pragma(msg, __traits(getAttributes, x)); // tuple()
+```
+
+Since memory safety-related attributes now have an effect on variables and fields, it becomes useful to inspect them.
+Therefor the restriction on the getFunctionAttributes trait gets lifted.
+
+The name "function attributes" is a bit unfortunate in this case, but this DIP does not aim to fix that.
+
+**(5) `bool` becomes an unsafe type**
+
+As explained in the [rationale](#rationale) section, `bool` types either cause memory corruption when assumed to be always 0 or 1, or might as well be `ubyte` types when almost every operation needs to truncate the value.
+By making `bool` an unsafe type, it can be both fast and `@safe` since all tricks to make invalid booleans are now disabled in `@safe` code.
+The cost is that existing code might break.
+Note that with the proposed rules a `bool` variable or field is only `@system` when the compiler cannot possibly prove that the initial value is 0 or 1, which should be a rare occurrence.
+If there is code breakage, it will likely come from overlap or void initialization of booleans.
+```D
+immutable ubyte ub = 3;
+bool getValidBool()   pure @system {return true;}
+bool getInvalidBool() pure @system {return *(cast(bool*) ub);}
```

Comment:
You probably meant ˋbool getInvalidBool() pure @system {return *(cast(bool*) &ub);}ˋ
(Note the `&`)